From 6db431911e3bee0e0d09211e5cdd3b85f4282066 Mon Sep 17 00:00:00 2001
From: Fugang Duan <B38611@freescale.com>
Date: Wed, 23 Apr 2014 15:12:41 +0800
Subject: [PATCH 1056/1074] ENGR00310027 net:fec: return the suitable value in
 irq handler

The current return value is not right beacause there may have irq
interrupt that is processed. So correct the return value.

Signed-off-by: Fugang Duan <B38611@freescale.com>
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 drivers/net/ethernet/freescale/fec_main.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/net/ethernet/freescale/fec_main.c b/drivers/net/ethernet/freescale/fec_main.c
index 55878b5..c7b57b8 100644
--- a/drivers/net/ethernet/freescale/fec_main.c
+++ b/drivers/net/ethernet/freescale/fec_main.c
@@ -1216,7 +1216,7 @@ fec_enet_interrupt(int irq, void *dev_id)
 
 	do {
 		if (unlikely(!fec_enet_collect_events(fep)))
-			return IRQ_NONE;
+			return ret;
 
 		if (fep->work_ts && fep->bufdesc_ex) {
 			ret = IRQ_HANDLED;
-- 
1.7.5.4

